### DISCUSSION TITLE
Raspberry Pi 4 + RPi OS Bookworm

### DISCUSSION THREAD

**Discussion by SvenVD** [OpenSurv/OpenSurv#22] [Hardware + Operating system]
All experiences and tests with OpenSurv on Raspberry Pi 4 + RPi OS Bookworm

(Raspberry Pi 4 + Ubuntu 24.04 LTS look here https://github.com/OpenSurv/OpenSurv/discussions/7)

**Comment by SvenVD** [comment]
issue with autostart https://github.com/OpenSurv/OpenSurv/issues/19

**Reply by Alex81m** [reply]
Hi, 
Hardware: Raspi 4 4 GB
OS: Bookworm Lite 64 bit
Git Pull today v 1.2
Installed with install.sh script
Unfortunately it seams that YAML Modul is not anymore installed with the script 
here the main.log
`2025/01/03 20:43:10 - DAEMON: Traceback (most recent call last):
2025/01/03 20:43:10 - DAEMON: File "/home/opensurv/lib/surveillance.py", line 7, in <module>
2025/01/03 20:43:10 - DAEMON: from core.util.config import cfg
2025/01/03 20:43:10 - DAEMON: File "/home/opensurv/lib/core/util/config.py", line 1, in <module>
2025/01/03 20:43:10 - DAEMON: import yaml
2025/01/03 20:43:10 - DAEMON: ModuleNotFoundError: No module named 'yaml'
`
Is this a known issue?

**Reply by Alex81m** [reply]
I now added pipx and than with pipx the pyyaml. Now it is starting

**Reply by chkpwd** [reply]
@Alex81m do you have RTSP working reliably on the Pi4? Mine stutter.

**Reply by chkpwd** [reply]
What I meant to ask was if you are displaying the cameras on a Raspberry Pi?

**Comment by arekmitura** [comment]
Hello,

Configuration:
Hardware: Raspi 4B 4 GB
OS: Bookworm 64 bit Release from 19.11.2024

1. Set up screen resolution to 1920x1080@60Hz
2. Update packages by -> `sudo apt-get install` 
3. Install python3-yaml -> `sudo apt install python3-yaml` 
4. Install OpenSurv ver. 1.3 standard following steps from description:
4a. `git clone https://github.com/OpenSurv/OpenSurv` 
4b. `cd OpenSurv`
4c. `git checkout v1_latest`
4d. `sudo ./install.sh`  -> overwrite OpenSurv config `yes` 

Result: Succesful  Autostart

In progress: Testing with rtsp streams

**Reply by Vattar** [reply]
> Hello,
> 
> Configuration: Hardware: Raspi 4B 4 GB OS: Bookworm 64 bit Release from 19.11.2024
> 
> 1. Set up screen resolution to 1920x1080@60Hz
> 2. Update packages by -> `sudo apt-get install`
> 3. Install python3-yaml -> `sudo apt install python3-yaml`
> 4. Install OpenSurv ver. 1.3 standard following steps from description:
>    4a. `git clone https://github.com/OpenSurv/OpenSurv`
>    4b. `cd OpenSurv`
>    4c. `git checkout v1_latest`
>    4d. `sudo ./install.sh`  -> overwrite OpenSurv config `yes`
> 
> Result: Succesful Autostart
> 
> In progress: Testing with rtsp streams

Forgive my ignorance, but how do I change the resolution to 1080p? My monitor is Full HD, and the system is already configured for the HDMI outputs to run at 4K 30Hz.

**Reply by winkelu** [reply]
Hello
I have still the probem that it boot to the CLI
I have done all 4 steps

for Info:
 cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 12 (bookworm)"
NAME="Raspbian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

any Ideas?

**Comment by arekmitura** [comment]
I tested the streaming by rstp today with 6 cameras using layout 1 configuration (from example) and works fine.
in h.264 stream is more stable - I didnt measured it but its look like fps are around 40-60 and its more smooth streaming -
in h.265 stream is sometime little unstable - also didnt measured but looks like its around 20-35 fps I assume it because of little tearring.
But I think it is not a opensurv problem just limitations of Raspberry Pi 4 and Ethernet connection.
And with h.264 sometimes on some of streams has loading screen appearing on few seconds but it also can be just problem with connection.

More inconvenient here is switching between defined screens sometime its almost immidiately but sometimes it takes 5-8 seconds and you are not sure it works :P

Overally I am satisfied how it works.

with screen buffering option enabled in general.yml it works very fine maybe a little more fluent

But I advise to use cooling fan for rpi :)

I know it is not a place for this but.... anybody know about how to add functionality for mouse buttons to switch the screens (next and previous)?

**Comment by arekmitura** [comment]
I have some problem with the boot of lightdm.service during startup

once I used startx command from cli or change in preferences run option mode CLI or DESKTOP and from that time it can't start automatically the lightdm.service. How to repair manualy systemd automatic boot up?

**Reply by SvenVD** [reply]
just a wild guess but try:

`mv -v /home/opensurv/.Xauthority /home/opensurv/.Xauthority.back`
`sudo systemctl restart lightdm` or `reboot`

**Comment by arekmitura** [comment]
Thank you for try but it had changed nothing

-------- Oryginalna wiadomość --------
7.02.2025 21:46, SvenVD  napisał(a):

**Comment by Wild1145** [comment]
I'm currently in the process of writing up a longer blog so I remember everything but in case it's helpful for anyone else stumbling across this thread until then. I had the same experience with my Pi 4 B where the initial install did fail and I needed to run a `sudo apt install python3-yaml` to get it fully up and running. Once I did that I was able to view the demo files and images without any issues. 

The other issue (and I'll submit a PR to fix the one line issue) was around the config example files in the Github repo being incorrect and referencing `camera_streams` not `streams` which caused me some issues. 

I did have ongoing issue with terrible performance and latency and tried a whole host of things from overriding resolutions to changing video output settings and hardware encoding settings but I haven't been able to find the right series of flags for `mpv` to make it work both with a real time view (rather than up to 2 min delays) or without terrible quality or frame-rates. It looks like the issue is the CPU on my Pi maxing out so I'm just going to chalk it up to not being powerful enough to do what I actually want it to do but from an installing this software perspective other than the already identified issue around the missing python-yaml package it seemed to work without issues.

**Comment by DigitalDreamer64** [comment]
I'm a former rpisurv user and just upgraded my rpi4 to a working Debian GNU/Linux 12 (bookworm) and 4-camera OpenSurv setup.

I initially had issues with getting this to run, so reverted back to Buster and rpsisurv. But, couldn't get rpsisurv to run again, either, due to latest VLC version not having any mmal related modules. Found a way to downgrade VLC to 3.0.17.4, but that didn't work. DEBUG log says to make sure VLC binary is running as root.

Decided to try RPi OS Bookworm again, but included the python3-yaml install as shown in this thread. Yay! Thanks, guys!
Happy with the result.

My big beefs don't appear to be related to OpenSurv at all. One is the fact that if I make any changes to the OS, the auto start on boot no longer works. I have to reinstall. Is there a better way to fix this? I know I can SSH into it and execute sudo systemctl start lightdm.service to get it going again, but I want it to start on boot.

The other problem with it is the fact that it doesn't shutdown properly, or at least like Buster did. I went and updated the OS and rpi-eeprom for the power switch to work, but no success. This isn't a OpenSurv issue, but a OS issue with the RPi 4 and power switch, I'm sure.

EDIT: What startup process is OpenSurv using to launch at boot time? (rc.local, .bashrc, init.d tab, systemd, crontab)
Or, do I need to add my own?

EDIT 2: Got the auto launch on boot working. The RPi needs to be set to boot to Desktop, not command line, for the autostart to work, as autostart only works for Desktop.

Thanks for all your work.
MAJ

**Reply by DigitalDreamer64** [reply]
The following is further follow-up information:

1.
Autostart is still not working. I just manually start it via command line or don't bother shutting it down.
EDIT: Autostart works now. I shutdown automatically via AppleScript on my Mac.

2.
The other issue is with one stream getting far behind. After several days it can be some 30 seconds behind, which I didn't think would be possible. Is there a buffer that is growing with time in this situation? DIdn't have this issue with rpisurv and Buster.

Three of the cameras are 1080P with a substream at 640x480 for OpenSurv, same as with rpisurv. The one that gets behind is a 3840x2160 camera with a substream at 704x480 with H.265. It has a wider aspect ratio, so I prefer this resolution. I have modified this substream to encode at H.264, but it still gets behind. I have just now switched to a 640x480 resolution and will monitor its behavior and report back. If there are any suggestions, let me know. 

And, BTW, all substreams are set at 15fps.

EDIT: After 5 hours, that one substream is 7 seconds behind.

EDIT 2: One quick way to get around the lagging videos is to simply switch to another screen and back. Video will get refreshed to beginning.

Thanks!
MAJ

**Comment by ajquick** [comment]
I’ve been testing on a Raspberry Pi 4B. With RPI Surv and three 4K streams, everything worked well. After switching to Open Surv and adding one or two more streams, it started crashing and hanging. I don’t think this generation of the Pi has enough memory or video processing power to handle large 4K streams.

**Reply by SvenVD** [reply]
Rpisurv was designed specifically for the Raspberry Pi and optimized to take full advantage of its GPU. The drawback is that it only works on Raspberry Pi devices.

Opensurv takes a different approach: it’s built to run on [a wide range of hardware](https://github.com/OpenSurv/OpenSurv/discussions/categories/hardware-operating-system), with varying levels of GPU support included. However, because the Raspberry Pi GPU is highly specialized for its own hardware, Opensurv can’t fully leverage it, which leads to lower performance on Pis.

As a result, on Raspberry Pi devices (up to the Pi 4), Rpisurv will always outperform Opensurv.

**Reply by ajquick** [reply]
That is good to know. I might give Rpisurv another try, otherwise I'll replace my hardware with something a bit more robust and stick with OpenSurv. I'm going to try an OrangePi, but it is possible that may suffer from similar hardware limitations? I haven't seen anyone mention OrangePi yet.

**Comment by tigertony21** [comment]
I've just set up a Rpi 4 (4gb) with Bookworm and now with Trixie together with the standard Raspberry 7" touchscreen. It's been running for a few days now. I'm only using this as a small monitor just to view my 2 door cameras (2K and 4K).  Everything is working fine with no issues and no lag whatsoever.

But the Touchscreen is on permanently. I really need it to turn off after a short time and then re awaken when touched  ?   This OpenSurv program seems to keep the display permanently on.
With other programs I run, the touchscreen times out as normal.
Any ideas ?

**Comment by Billabomb75** [comment]
So found a solution with the login screen issue. That worked for me.  
Found that lightdm.service is in a dead state upon startup.  via `systemctl status lightdm.service` 
I first checked the display manager. 

`cat /etc/X11/default-display-manager` should output `/usr/sbin/lightdm`

What I did that worked is I removed the default.target file and set the default graphical.target. Restarted and now logs in no problem and displays my cameras properly as per the monitor1.yml file. 
`rm /etc/systemd/system/default.target`
`systemctl set-default graphical.target`

"This will replace the default target file, which in my case must have prevented systemd from reaching the graphical target, with a symlink to the graphical target. After this, lightdm (and gdm when selected with dpkg-reconfigure) starts automatically."

Thanks to askubuntu https://askubuntu.com/a/939995

**Reply by bitvilag** [reply]
had the same issue on trixie with pi4b
